### PR TITLE
1kLoC Enhancement: Improve find

### PR DESCRIPTION
**Pull request by WeeSooJun**
@Happytreat

### CODE REVIEW

**Review by Happytreat — COMMENTED** [review]
Great use of enum in this case to make the code clearer! One thing I have noticed: 

- Use present tense verb when writing commit message: 
For instance: "Add PersonProperty Enum and change Person data type"

Also, maybe you can also think about how changing the String[] to HashMap<> affects the time complexity of your code (especially if we have to add many persons into the addressbook). 
@WeeSooJun

**Comment by WeeSooJun** [comment]
Alright, I will take note of the use of present tense for future commits.

Regarding the time complexity:

1. By using a hashmap to represent and store a person's details. The put and get operation should be O(1) if keys are even distributed which may not always be the case. 

2. Since the program has fixed number of details about a person, there would not be a need to add different details to an array avoiding the O(n) scenario of adding into an already full array.

So in conclusion the change from String[] to Hashmap may slow the program. Is this true?

Thanks!
@Happytreat

**Comment by kylase-learning** [comment]
Remember to close this PR after reading it. :)
